osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/datadecoder/dlmsclassdatadecoder/TimeAndEventsClassesDecoder.java

```diff
@@ -34,6 +38,33 @@ public TimeAndEventsClassesDecoder(
     this.basicDlmsDataDecoder = basicDlmsDataDecoder;
   }
 
+  // Class-id 8: Clock
+  public String decodeClockStatus(final DataObject attributeData) {
+    try {
+      final int value = this.dlmsHelper.readInteger(attributeData, "read clock status");
+      final Set<ClockStatusBitDto> statusBits = ClockStatusBitDto.forClockStatus(value);
+      if (statusBits != null && !statusBits.isEmpty()) {
+        return "status: "
+            + statusBits.stream()
+                .map(ClockStatusBitDto::getDescription)
+                .collect(Collectors.joining(", "));
+      } else {
+        return "no status set";
+      }
+    } catch (final Exception e) {
+      return "decoding clock base failed: " + e.getMessage();
```

Comment:
Does this exception message contain the raw data? If not perhaps idea to add it to the return string?

Reply:
Exception is caught and raw data is added in other class